src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -4,12 +4,92 @@
 
 public class StorageImpl<K, V> implements Storage<K, V> {
 
+    private static final int MAXIMUM_SIZE = 10;
+    private final Pair<K, V>[] pairsArray;
+    private int currentAmount;
+
+    public StorageImpl() {
+        pairsArray = new Pair[MAXIMUM_SIZE];
+    }
+
+    public int getCurrentAmount() {
+        return this.currentAmount;
+    }
+
     @Override
     public void put(K key, V value) {
+        Pair<K, V> input = Pair.of(key, value);
+
+        for (int i = 0; i < currentAmount; i++) {
+            if (pairsArray[i] != null
+                    && (pairsArray[i].first == key
+                    || key != null && key.equals(pairsArray[i].first))) {
+                pairsArray[i] = input;
+                return;
+            }
+        }
+        pairsArray[currentAmount] = input;
+        currentAmount++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < currentAmount; i++) {
+            Pair<K, V> pair = pairsArray[i];
+            if (pair != null
+                    && ((pair.first == key)
+                    || (key != null && key.equals(pair.first)))) {
+                return pair.second;
+            }
+        }
         return null;
     }
+
+    private static class Pair<K, V> {
+        private static final int PRIME = 97;
+        private final K first;
+        private final V second;
```

Comment:
```suggestion
        private K first;
        private V second;
```
what is we decide to create setters?